### PR TITLE
Update expectations for uninitialized internal function.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 1]
```diff
@@ -5626,11 +5626,20 @@ BOOST_AUTO_TEST_CASE(uninitialized_internal_storage_function)
...
+		compileAndRun(sourceCode, 0, "Test");
 
-	bytes result = callContractFunction("f()");
-	BOOST_CHECK(!result.empty());
-	BOOST_CHECK(result != encodeArgs(0));
+		bytes result = callContractFunction("f()");
+		BOOST_CHECK(!result.empty());
+		// Via yul, the uninitialized function pointer is zero, because
+		// that index does not appear in the dispatch function.
+		// The old code generator has an explicit "invalid" internal function.
```
Comment (axic) [inline test/libsolidity/SolidityEndToEndTest.cpp]: Any reason for this? This should be also added to `docs/ir/ir-breaking-changes.rst`

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 2]
```diff
@@ -5626,11 +5626,20 @@ BOOST_AUTO_TEST_CASE(uninitialized_internal_storage_function)
 			}
 		}
 	)";
-	compileAndRun(sourceCode, 0, "Test");
+	ALSO_VIA_YUL(
```
Comment (axic) [inline test/libsolidity/SolidityEndToEndTest.cpp]: Any chance we could move this into semantic tests? Would require two separate files, one with `compileViaYul: false` and the other with `true`.

Comment (mijovic) [reply test/libsolidity/SolidityEndToEndTest.cpp]: I would vote for moving them to `semanticTests` with two files.

Comment (chriseth) [reply test/libsolidity/SolidityEndToEndTest.cpp]: The problem is that the identifier returned by the old codegen will be flaky and change with optimizer changes. I can move the check into the code, though, if you like.

Comment (mijovic) [reply test/libsolidity/SolidityEndToEndTest.cpp]: I am fine with leaving it here in that case

**Comment by mijovic** [comment]
This looks good apart from open point about documenting in `docs/ir/ir-breaking-changes.rst`

**Review by mijovic — APPROVED** [review]

**Pull request merged**